irods/test/data_obj_test.py

```diff
@@ -2094,6 +2105,132 @@ def test_touch_operation_does_not_work_when_given_a_collection__525(self):
         with self.assertRaises(ex.InvalidInputArgument):
             user_session.data_objects.touch(home_collection_path)
 
+    @unittest.skipIf(progressbar is None, "progressbar is not installed")
+    def test_pbar_for_parallel_io_4(self):
+        from irods.manager.data_object_manager import (register_update_instance, _update_fns)
+
+        class wrapper:
+            def __init__(self,pbar):
+                self.pbar = pbar
+                self.lock = threading.Lock()
+                self.total = 0
+
+            def update(self,n):
+                with self.lock:
+                    self.total += n
+                    self.pbar.update(self.total)
+
+        LEN = 1024**2*40
+        content = b'_'*LEN
+
+        progress_bar = progressbar.ProgressBar(maxval=len(content))
+        wrapped = wrapper(progress_bar)
+
+        register_update_instance(wrapped, wrapped.update)
+
+        wrapped.pbar.start()
+        self._run_pbars_for_parallel_io(content, [wrapped])
+        self.assertEqual(wrapped.pbar.currval, LEN)
+
+    @unittest.skipIf(progressbar is None, "progressbar is not installed")
+    def test_pbar_for_parallel_io_3(self):
+        from irods.manager.data_object_manager import (register_update_type, unregister_update_type)
+
+        class ProgressBar_wrapper:
+         def start(self,*x):return self.p.start(*x)
+         def update(self,*x):return self.p.update(*x)
+         @property
+         def currval(self): return self.p.currval
+         def __init__(self,*args,**kw):
+          self.p = progressbar.ProgressBar(*args,**kw)
```

Comment:
Please make the formatting consistent here. I think I'd prefer things on separate lines with indentation.